WcaOnRails/app/assets/javascripts/admin.js

```diff
@@ -0,0 +1,30 @@
+onPage('admin#edit_person, admin#update_person', function() {
+  var $personFields = $('#person-fields :input');
+
+  $('#person_wca_id').on('change', function() {
+    if($(this).val() === '') {
+      $personFields.attr('disabled', true);
+      $personFields.not('[type="submit"]').val('');
+    } else {
+      $personFields.attr('disabled', false);
+      wca.cancelPendingAjaxAndAjax('grab_person_data', {
+        url: '/admin/person_data',
+        data: {
+          'person_wca_id': $(this).val(),
+        },
+        success: function(data) {
+          $.each(data, function(attribute, value) {
+            $('#person_' + attribute).val(value);
+            if(attribute === 'dob') {
+              $('#person_dob').data("DateTimePicker").date(value);
+            }
+          });
+        }
+      });
+    }
+  });
+
+  if($('#person_wca_id').val() === '') {
```

Comment:
Yeah, it's populated as you described.
I've added this because if you pick a person all the other fields are filled in but when you refresh they stay on their places apart from the wca_id. Isn't it about this user_ids picker?